src/wp-includes/class-wp-image-editor-gd.php

```diff
@@ -570,4 +570,62 @@ protected function make_image( $filename, $callback, $arguments ) {
 
 		return parent::make_image( $filename, $callback, $arguments );
 	}
+
+	/**
+	 * Get dominant color from a file.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @return string|WP_Error Dominant hex color string, or an error on failure.
+	 */
+	public function get_dominant_color() {
```

Comment:
Need to add these methods to the `WP_Image_Editor` class.

Reply:
Second thoughts maybe this is not the best idea. Let look at https://github.com/WordPress/performance/pull/404